processor/stream/stream_processor.go

```diff
@@ -36,6 +39,10 @@ import (
 	"github.com/elastic/apm-server/validation"
 )
 
+const (
+	RateLimiterKey = "rateLimiter"
```

Comment:
It's not likely to be an issue, but it's best practice to use an unexported struct type as a context key, to avoid clashing with keys from other code. So this would be something like:

```go
type rateLimiterKey struct {}
var RateLimiterKey = rateLimiterKey{}
```

or perhaps in order to avoid an exported var:

```go
type rateLimiterKey struct {}

func ContextWithRateLimiter(ctx context.Context, limiter *rate.Limiter) context.Context {
    return context.WithValue(ctx, rateLimiterKey{}, limiter)
}
```

Reply:
We've used a string wrapped in a context key type here: https://github.com/elastic/apm-server/blob/master/beater/handlers.go#L311 and a `ContextWith` function here: https://github.com/elastic/apm-server/blob/v2/utility/request_time.go#L29

would probably make sense to align the code base on how we do it. I have no strong feelings of which approach as long as it's the same.